test/oracle.test.ts

```diff
@@ -15,9 +16,57 @@ contract('Oracle', ([owner]) => {
     oracle = await Oracle.new({from: owner})
   })
 
-  describe('oracle', () => {
-    it('oracle', async () => {
-      //empty
+  describe('Batch oracle', () => {
+    it('should revert setting batch oracle from non-owner address', async () => {
+      await expectRevert(oracle.setBatchOracle(batch, batchOracle, {from: random}), 'Ownable: caller is not the owner')
+    })
+
+    it('should set batch oracle', async () => {
+      await oracle.setBatchOracle(batch, batchOracle, {from: owner})
+
+      assert.equal(await oracle.getBatchOracle(batch), batchOracle, 'batch oracle address mismatch')
+    })
+  })
+
+  describe('Oracle locking period', () => {
+    const lockingPeriod = new BigNumber(60 * 15) // 15min
+
+    it('should revert setting oracle locking period from non-owner address', async () => {
+      await expectRevert(
+        oracle.setLockingPeriod(batchOracle, lockingPeriod, {from: random}),
+        'Ownable: caller is not the owner',
+      )
+    })
+
+    it('should set oracle locking period', async () => {
+      await oracle.setLockingPeriod(batchOracle, lockingPeriod, {from: owner})
+
+      assert.equal(
+        (await oracle.getOracleLockingPeriod(batchOracle)).toString(),
+        lockingPeriod.toString(),
+        'oracle locking period mismatch',
+      )
+    })
+  })
+
+  describe('Oracle dispute period', () => {
+    const disputePeriod = new BigNumber(60 * 45) // 45min
+
+    it('should revert setting oracle locking period from non-owner address', async () => {
+      await expectRevert(
+        oracle.setDisputePeriod(batchOracle, disputePeriod, {from: random}),
+        'Ownable: caller is not the owner',
+      )
+    })
+
+    it('should set oracle locking period', async () => {
+      await oracle.setDisputePeriod(batchOracle, disputePeriod, {from: owner})
+
+      assert.equal(
```

Comment:
Again, probably good to have some upper and lower bounds on dispute period. 

Are the dispute and locking periods set once or set per oracle? I’m a little confused by that.

Reply:
yeah locking and dispute period are set per oracle in this case.

Reply:
maybe 3 days max, 15 minutes min? (by oracle you mean per strike-expiry-underliying-collateral? )

Reply:
a locking and dispute period is set by oracle address.